web/src/features/manual-scoring/components/AnnotateDrawer.tsx

```diff
@@ -301,47 +303,67 @@ export function AnnotateDrawer({
-        ({ label }) => label === stringValue,
-      );
-      if (selectedCategory) {
-        const newValue = Number(selectedCategory.value);
-
-        update(index, {
-          ...score,
-          value: newValue,
-        });
+      try {
+        const selectedCategory = configCategories.find(
+          ({ label }) => label === stringValue,
+        );
+        if (selectedCategory) {
+          const newValue = Number(selectedCategory.value);
+
+          update(index, {
+            ...score,
+            value: newValue,
+          });
 
-        if (!!stringValue) {
-          if (!!score.scoreId) {
-            await mutUpdateScores.mutateAsync({
-              projectId,
-              ...score,
-              id: score.scoreId,
-              value: newValue,
-              stringValue,
-            });
-            capture("score:update", {
-              type: type,
-              source: source,
-              dataType: score.dataType,
-            });
-          } else {
-            await mutCreateScores.mutateAsync({
-              projectId,
-              traceId,
-              ...score,
-              observationId,
-              value: newValue,
-              stringValue,
-            });
-            capture("score:create", {
-              type: type,
-              source: source,
-              dataType: score.dataType,
-            });
+          if (!!stringValue) {
+            if (!!score.scoreId) {
+              const validatedScore = UpdateAnnotationScoreData.parse({
+                id: score.scoreId,
+                projectId,
+                traceId,
+                name: score.name,
+                dataType: score.dataType,
+                configId: score.configId,
+                comment: score.comment,
+                observationId,
+                value: newValue,
+                stringValue,
+              });
+
+              await mutUpdateScores.mutateAsync({
+                ...validatedScore,
+              });
+              capture("score:update", {
+                type: type,
+                source: source,
+                dataType: score.dataType,
+              });
+            } else {
+              const validatedScore = CreateAnnotationScoreData.parse({
+                projectId,
+                traceId,
+                name: score.name,
+                dataType: score.dataType,
+                configId: score.configId,
+                comment: score.comment,
+                observationId,
+                value: newValue,
+                stringValue,
+              });
+
+              await mutCreateScores.mutateAsync({
+                ...validatedScore,
+              });
+              capture("score:create", {
+                type: type,
+                source: source,
+                dataType: score.dataType,
+              });
+            }
           }
         }
+      } catch (error) {
+        trpcErrorToast(error);
```

Comment:
trpcErrorToast(error) does not need to be passed manually or onError above, it is used automatically as onError for all trpc routes unless it is overridden by onError of a mutation or query